src/content/app/genome-browser/state/track-config/trackConfigSlice.ts

```diff
@@ -68,52 +98,190 @@ export const updateTrackStatesAndSave: ActionCreator<
   browserStorageService.saveTrackStates(trackStates);
 };
 
+export const getTracksStateInfo = (
+  genomeId: string,
+  trackCategories: GenomeTrackCategory[]
+) => {
+  const tracksInfo: TrackConfigs = {};
+
+  tracksInfo['gene-focus'] = {
+    showSeveralTranscripts: false,
+    showTranscriptIds: false,
+    showTrackName: false,
+    showFeatureLabel: true,
+    trackType: TrackType.GENE
+  };
+
+  trackCategories.forEach((category) => {
+    category.track_list.forEach((track) => {
+      const trackId = track.track_id.replace('track:', '');
+      const trackType = getTrackType(trackId);
+      if (trackType === TrackType.GENE) {
+        tracksInfo[trackId] = {
+          showSeveralTranscripts: false,
+          showTranscriptIds: false,
+          showTrackName: false,
+          showFeatureLabel: true,
+          trackType
+        };
+      } else {
+        tracksInfo[trackId] = {
+          showTrackName: false,
+          trackType
+        };
+      }
+    });
+  });
+  return { tracks: tracksInfo };
+};
+
+export const getTrackConfigsPerGenome = (
+  genomeId: string,
+  trackCategories: GenomeTrackCategory[]
+): TrackConfigsPerGenome => {
+  return genomeId
+    ? {
+        ...defaultTrackConfigsPerGenome,
+        ...getTracksStateInfo(genomeId, trackCategories),
+        ...getPersistentTrackConfigsPerGenome(genomeId)
+      }
+    : defaultTrackConfigsPerGenome;
+};
+
+export const getPersistentTrackConfigsPerGenome = (
+  genomeId: string
+): Partial<TrackConfigsPerGenome> => {
+  const trackConfigState =
+    browserTrackConfigStorageService.getTrackConfigState();
+  if (!genomeId || !trackConfigState[genomeId]) {
+    return {};
+  }
+  return trackConfigState[genomeId] || {};
+};
+
 const browserTrackConfigSlice = createSlice({
   name: 'genome-browser-track-config',
-  initialState: defaultTrackConfigState,
+  initialState: {} as GenomeTrackConfigs,
   reducers: {
-    updateCogList(state, action: PayloadAction<number>) {
-      state.browserCogList = action.payload;
-    },
-    updateCogTrackList(state, action: PayloadAction<CogList>) {
-      state.browserCogTrackList = action.payload;
-    },
-    updateSelectedCog(state, action: PayloadAction<string | null>) {
-      state.selectedCog = action.payload;
+    setInitialTrackConfigsPerGenome(
+      state,
+      action: PayloadAction<{
+        genomeId: string;
+        trackCategories: GenomeTrackCategory[];
+      }>
+    ) {
+      const { genomeId, trackCategories } = action.payload;
+      if (!state[genomeId]) {
+        state[genomeId] = getTrackConfigsPerGenome(genomeId, trackCategories);
+      }
+      browserTrackConfigStorageService.setTrackConfigState({
+        genomeId: genomeId,
+        fragment: state[genomeId]
+      });
```

Comment:
Here and throughout the rest of the slice — I am very conflicted about writing data to browser storage from inside the reducers. Reducers are supposed to be pure functions, while interacting with the browser storage is a side effect. Not quite sure yet what to do with this; but it's worth thinking about.